marklogic-data-hub/src/test/java/com/marklogic/hub/impl/ClearUserModulesTest.java

```diff
@@ -0,0 +1,126 @@
+import com.marklogic.client.admin.TransformExtensionsManager;
+import com.marklogic.client.io.JacksonHandle;
+import com.marklogic.client.io.QueryOptionsListHandle;
+import com.marklogic.hub.AbstractHubCoreTest;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Assumptions;
+import org.junit.jupiter.api.Test;
+
+class ClearUserModulesTest extends AbstractHubCoreTest {
+
+    @Test
+    void testClearUserModules() {
+        new DataHubImpl(runAsDataHubDeveloper()).clearUserModules();
+        Assumptions.assumeTrue(isVersionCompatibleWith520Roles());
+        ModuleCounts beforeClearingModules = new ModuleCounts();
+        beforeClearingModules.setModuleCounts();
+
+        Assertions.assertEquals(1, beforeClearingModules.optionsCount);
+        Assertions.assertEquals(12, beforeClearingModules.transformsCount);
+        Assertions.assertEquals(23, beforeClearingModules.resourceExtensionCount);
+
+        installProjectWithEachTypeOfModule();
+
+        try {
+            new DataHubImpl(runAsDataHubOperator()).clearUserModules();
+            Assertions.fail("'data-hub-operator' should not be able delete user modules");
+        } catch (RuntimeException e) {
+            Assertions.assertTrue(e.getMessage().contains("User is not allowed to delete /config/query"));
+            logger.info("'data-hub-operator' cannot delete user modules");
+        }
+
+        ModuleCounts afterClearingModulesAsOperator = new ModuleCounts();
+        afterClearingModulesAsOperator.setModuleCounts();
+
+        Assertions.assertEquals(2, afterClearingModulesAsOperator.optionsCount);
+        Assertions.assertEquals(13, afterClearingModulesAsOperator.transformsCount);
+        Assertions.assertEquals(24, afterClearingModulesAsOperator.resourceExtensionCount);
+        Assertions.assertEquals(beforeClearingModules.hubCoreModuleCount, afterClearingModulesAsOperator.hubCoreModuleCount);
+        checkIfCustomModulesArePresent();
+
+        new DataHubImpl(runAsDataHubDeveloper()).clearUserModules();
+
+        ModuleCounts afterClearingModulesAsDeveloper = new ModuleCounts();
+        afterClearingModulesAsDeveloper.setModuleCounts();
+
+        Assertions.assertEquals(beforeClearingModules.optionsCount, afterClearingModulesAsDeveloper.optionsCount);
+        Assertions.assertEquals(beforeClearingModules.transformsCount, afterClearingModulesAsDeveloper.transformsCount);
+        Assertions.assertEquals(beforeClearingModules.resourceExtensionCount, afterClearingModulesAsDeveloper.resourceExtensionCount);
+        Assertions.assertEquals(beforeClearingModules.hubCoreModuleCount, afterClearingModulesAsDeveloper.hubCoreModuleCount);
+        //Before test starts , additional custom modules may be present
+        Assertions.assertTrue(afterClearingModulesAsDeveloper.totalModulesCount <= beforeClearingModules.totalModulesCount);
```

Comment:
For assertTrue/assertFalse, it's often helpful to have an assertion message that shows the actual values; otherwise JUnit just says that we didn't get true back, which makes debugging more difficult. We can remove the comment in favor of an assertion message too - e.g. "Expected total module count to be same or less than the before count, as some custom modules may have existed from other tests; before count: " + beforeCount + "; after count: " + afterCount

Generally, assertion messages are better at explaining things than comments, and they're more likely to stay in sync with the code.